geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geohash/GeohashUtils.scala

```diff
@@ -763,18 +763,20 @@ object GeohashUtils
       }
     }
 
+    // copy the geometry so that if we fail it doesn't get messed up
+    val copy = defaultGeometryFactory.createGeometry(targetGeom)
     val withinBoundsGeom =
       if (targetGeom.getEnvelopeInternal.getMinX < -180 || targetGeom.getEnvelopeInternal.getMaxX > 180)
-        translateGeometry(targetGeom)
+        translateGeometry(copy)
       else
-        targetGeom
+        copy
 
     try {
       JtsSpatialContext.GEO.makeShape(withinBoundsGeom, true, true).getGeom
     } catch {
       case e: Exception =>
         logger.warn(s"Error splitting geometry on IDL for $withinBoundsGeom", e)
-        withinBoundsGeom
+        targetGeom
```

Comment:
My concern is the method (getInternationalDateLineSafeGeometry) is basically trying to do something and then if it fails for any reason, gives up and returns the original geom.  If we can just return the original copy on any failure, then why call this method in the first place.  You can make an argument that a 'warn' gets logged, but those generally get lost in practice and potentially kicks the can down the road to another error that is even harder to debug bc one assumes this method did it's job.  I'd really prefer we just throw (assuming this method has value) since the problem is the caller needs to fix their polygon (assuming geomesa really only wants ones with edges < 180, which seems kind of silly since the guy writing that spatial4j library basically said he implemented it this way because it was easier than following the standard and he didn't think anyone used polygons that big in practice).